### PR TITLE
CURA-2713: NSIS: Updating commands for Arduino drivers

### PR DESCRIPTION
**Pull request by thopiekar**
Updates commands for the updated Arduino drivers in https://github.com/Ultimaker/cura-binary-data/pull/5

_Includes_:
Fixes #89
Fixes #92
Fixes #93

_Depends on_:
https://github.com/Ultimaker/cura-binary-data/pull/5

Contributes to CURA-2713

### CODE REVIEW

File: cmake/NSIS.template.in [hunk 1]
```diff
@@ -937,9 +937,123 @@ SectionEnd
...
+
+uninst_23:
+  ClearErrors
+  StrLen $2 "\Uninstall.exe"
+  StrCpy $3 $0 -$2 # remove "\Uninstall.exe" from UninstallString to get path
+  ExecWait '$0 _?=$3' ;Do not copy the uninstaller to a temp file
+
+  IfErrors uninst_failed lookForCurrentVersion
+
+lookForCurrentVersion:
```
Comment (Ghostkeeper) [inline cmake/NSIS.template.in]: Several releases are still missing.

I would also hope that we won't need to modify this code every time we make a release though. Currently it seems to work automatically. Why was this necessary for updating the Arduino drivers?

File: packaging/windows.cmake [hunk 2]
```diff
@@ -60,7 +60,8 @@ set(CPACK_RESOURCE_FILE_LICENSE ${CMAKE_SOURCE_DIR}/LICENSE)
 set(CPACK_PACKAGE_CONTACT "Arjen Hiemstra <a.hiemstra@ultimaker.com>")
 
 set(CPACK_PACKAGE_EXECUTABLES Cura "Cura ${CURA_VERSION_MAJOR}.${CURA_VERSION_MINOR}.${CURA_VERSION_PATCH}")
-set(CPACK_PACKAGE_INSTALL_DIRECTORY "Cura ${CURA_VERSION_MAJOR}.${CURA_VERSION_MINOR}")
+set(CPACK_PACKAGE_INSTALL_DIRECTORY "Cura")
+set(CPACK_PACKAGE_INSTALL_REGISTRY_KEY "Cura")
```
Comment (Ghostkeeper) [inline packaging/windows.cmake]: This prevents users from installing multiple versions alongside each other. I know it's not common, but historically this is what our user base wants to be able to do, so I don't think we should change that right now.

**Comment by awhiemstra** [comment]
Note that a bunch of these changes have already been merged as they were part of different pull requests, among others the version detection changes. This PR should really be rebased on master first (or maybe even just recreated).

**Comment by thopiekar** [comment]
@awhiemstra Are you sure? When I installed the 2.7 beta I noticed that the Arduino drivers were unsigned. But from what I remember is that the drivers I uploaded to cura-binary-data, these were signed and should prevent Windows' warning message.

File: packaging/windows.cmake (hunk 2 above)
Comment (thopiekar) [reply packaging/windows.cmake]: Installing multiple versions is wanted? I better have one version installed, which works, instead of multiple ones, which are all working good in their own way..

File: cmake/NSIS.template.in (hunk 1 above)
Comment (thopiekar) [reply cmake/NSIS.template.in]: I don't know. It is the same question like: Why shall I install updates and upgrades my computer? :roll_eyes:

File: NSIS/CharStrStrip.nsh [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+Pop $R1
+Exch $R0
+FunctionEnd
+!macro StrStrip Str InStr OutVar
+ Push '${InStr}'
+ Push '${Str}'
+  Call StrStrip
+ Pop '${OutVar}'
+!macroend
+!define StrStrip '!insertmacro StrStrip'
```
[the remark this answers is not in the document]
Comment (thopiekar) [reply NSIS/CharStrStrip.nsh]: Hmm, I would need to look deeper into the code again. Currently have no time.

File: projects/cura-binary-data.cmake [hunk 4]
```diff
@@ -1,7 +1,7 @@
 
 ExternalProject_Add(cura-binary-data
-    GIT_REPOSITORY https://github.com/ultimaker/cura-binary-data
-    GIT_TAG origin/${TAG_OR_BRANCH}
+    GIT_REPOSITORY https://github.com/thopiekar/cura-binary-data
+    GIT_TAG origin/master-CURA-2713-clean
```
[the remark this answers is not in the document]
Comment (thopiekar) [reply projects/cura-binary-data.cmake]: Yep, just some leftovers.

**Comment by LipuFei** [comment]
And some merge conflicts to resolve.

**Comment by thopiekar** [comment]
Updated the files in cura-binary-data. Do you need some work here, too?`

**Comment by thopiekar** [comment]
Rebased the needed changes in #140
